### PR TITLE
Add basic websockets server

### PR DESCRIPTION
**Pull request by simison**
Proof of concept of a basic WebSocket server implementation.

The use case would be for the native mobile app to be able to pull via sockets instead of doing constant polling of the API.

#### Proposed Changes

* Work in progress

#### Testing Instructions

* Work in progress

### CODE REVIEW

File: modules/pages/client/components/Rules.component.js [hunk 1]
```diff
@@ -1,39 +1,15 @@
...
-          </div>
-        </div>
-      </Board>
-
       <section className="container container-spacer">
         <div className="row">
           <div className="col-xs-12 col-sm-offset-1 col-sm-10 col-md-offset-2 col-md-8">
             <div>
-              <RulesText />
+              <WebSocketDemo />
```
Comment (simison) [inline modules/pages/client/components/Rules.component.js:11]: Haha, needs moving to a better place. This was here just for a demo.

File: config/lib/websocket.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+          callback();
+        });
+      });
+    });
+  });
+}
+
+module.exports.start = function (app, connection) {
+  const map = new Map();
```
Comment (rastapasta) [inline config/lib/websocket.js:72]: would recommend using an Array instead to allow multiple connections (like web + app)

Comment (simison) [reply config/lib/websocket.js:72]: Yup 👍 makes sense. Feel free to push changes to this branch!

File: package.json [hunk 3]
```diff
@@ -171,6 +171,7 @@
     "react-map-gl": "5.2.7",
     "react-medium-editor": "1.8.1",
     "react-responsive": "8.1.0",
+    "react-use-websocket": "2.0.3-b.4",
```
Comment (simison) [inline package.json:174]: Did a cursory look at different options and nothing seemed to be at very established state. Not too tied to this specific one either.

**Pull request closed without merging**